packages/astro/src/integrations/index.ts

```diff
@@ -62,39 +62,39 @@ export async function runHookConfigDone({ config }: { config: AstroConfig }) {
 export async function runHookServerSetup({ config, server }: { config: AstroConfig; server: ViteDevServer }) {
 	for (const integration of config.integrations) {
 		if (integration.hooks['astro:server:setup']) {
-			await integration.hooks['astro:server:setup']({ server });
+			await integration.hooks['astro:server:setup']({ config, server });
```

Comment:
Looks like these `config` variables that you added aren't actually used by the integration. I think the intended pattern is, if you need the vite config later add a hook for `astro:config:done` and save the config as a variable.